src/Uno.SourceGeneration/GeneratorExecutionContext.cs

```diff
@@ -63,5 +63,12 @@ protected GeneratorExecutionContext(Compilation compilation, ParseOptions parseO
 		/// Unused, present for compatibilty
 		/// </summary>
 		public abstract void ReportDiagnostic(Diagnostic diagnostic);
+
+		public abstract bool TryGetValueOptionValue(SyntaxTree tree, string key, out string value);
+
+		public abstract bool TryGetValueOptionValue(AdditionalText textFile, string key, out string value);
+
+		public abstract bool TryGetGlobalOptionValue(string key, out string value);
```

Comment:
This doesn't match the API surface of Roslyn.

In Roslyn, `GeneratorExecutionContext` has `AnalyzerConfigOptionsProvider` property. But `AnalyzerConfigOptionsProvider` is Roslyn 3.1.0 which we cannot use here.
So, I added methods for the common uses of `AnalyzerConfigOptionsProvider`, and implemented them in `Uno.SourceGeneration.Host` which uses a Roslyn version higher than 3.1.0 and can access `AnalyzerConfigOptionsProvider`.

For consistent API usages, we'll have to introduce these three methods as extensions in Uno when we're using Roslyn generators.

Reply:
That's fair. Is there a way to validate that ?

Reply:
I'll look into that.